src/libbson/src/bson/bson.c

```diff
@@ -3271,16 +3274,17 @@ _bson_as_json_visit_all (const bson_t *bson,
          *length = 3;
       }
 
-      return bson_strdup ("{ }");
+      return bson_strdup (is_outermost_array ? "[ ]" : "{ }");
    }
 
    if (!bson_iter_init (&iter, bson)) {
       return NULL;
    }
 
    state.count = 0;
-   state.keys = true;
-   state.str = bson_string_new ("{ ");
+   state.keys = !is_outermost_array;
+   state.str =
+      is_outermost_array ? bson_string_new ("[ ") : bson_string_new ("{ ");
```

Comment:
```suggestion
   state.str = bson_string_new (is_outermost_array ? "[ " : "{ ");
```
Verbosity reduction.